ai.py

```diff
@@ -218,7 +265,9 @@ def pick_next_move(self, active_player_sign : int):
                     # If not at last layer, advance layer
                     moves[ply] = generate_moves(self.board, player_sign * (-1)**ply)
                     mptr[ply] = 0
-                    if ply > 1:
+                    if len(moves[ply]) == 0:
+                        scores[ply] = evaluate_position(self.board)
```

Comment:
If there are no moves that can be made, write the score of the current part of the tree